frontend/src/components/doctor/NewMedicalRecordMain.jsx

```diff
@@ -0,0 +1,122 @@
+import { useNavigate } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+import { fetchUserId } from "../../api/userFunctions";
+import { Tabs, Tab, Card, CardBody, Spinner, addToast } from "@heroui/react";
+import PatientSelection from "./PatientSelection";
+import NewMedicalRecordForm from "./NewMedicalRecordForm";
+
+const tabSteps = ["patient-selection", "record-submission"];
+
+export default function NewMedicalRecordMain() {
+  const { user, isLoading } = useAuth0();
+
+  //Doctors id
+  const [doctorId, setDoctorId] = useState();
+
+  //Set selected tab
+  const [selectedStep, setSelectedStep] = useState("patient-selection");
+
+  //Set Selected Patient
+  const [selectedPatient, setSelectedPatient] = useState();
+
+  //Set new medical record body
+  const [newMedicalRecord, setNewMedicalRecord] = useState({});
+
+  //Gets the doctors id
+  useEffect(() => {
+    if (!isLoading && user) {
+      const getUserId = async () => {
+        const id = await fetchUserId(user.email);
+        setDoctorId(id);
+      };
+      getUserId();
```

Comment:
Why do we need to create this async function `getUserId ` rather than just executing the wrapped code directly?